modules/services/gpg-agent.nix

```diff
@@ -71,7 +64,15 @@ in
     };
   };
 
-  config = mkIf cfg.enable (mkMerge [
+  config = let
+    gpgInitStr = ''
+        export GPG_TTY=$(tty)
+      '' + (optionalString cfg.enableSshSupport ''
+        ${pkgs.gnupg}/bin/gpg-connect-agent --quiet updatestartuptty /bye > /dev/null
```

Comment:
Why is this only added when `enableSshSupport == true`. Doesn't it also apply for PGP keys?

Reply:
from `gpg-agent` man page:

> Note:  in  case  the gpg-agent receives a signature request, the
>              user might need to be prompted for a passphrase, which is neces‐
>             sary  for decrypting the stored key.  Since the ssh-agent proto‐
>              col does not contain a mechanism for telling the agent on  which
 >             display/terminal it is running, gpg-agent's ssh-support will use
  >            the TTY or X display  where  gpg-agent  has  been  started.   To
   >           switch  this  display  to the current one, the following command
  >          may be used:
>
  > gpg-connect-agent updatestartuptty /bye

hence, no.

Reply:
Apologies for the delay, I forgot about this comment! Your point is very good. I agree it should be only be used when SSH support is needed.